swap/src/asb/event_loop.rs

```diff
@@ -318,16 +318,30 @@ where
         &mut self,
         min_buy: bitcoin::Amount,
         max_buy: bitcoin::Amount,
+        monero_wallet: Arc<monero::Wallet>,
```

Comment:
No need to pass this as a parameter, you have it on `self`. (Also visible by the fact that you pass `self.monero_wallet` into this function!)